backend/src/main/java/me/performancereservation/api/RefundController.java

```diff
@@ -0,0 +1,72 @@
+package me.performancereservation.api;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import me.performancereservation.domain.refund.RefundService;
+import me.performancereservation.domain.refund.dto.RefundDetailResponse;
+import me.performancereservation.domain.refund.dto.RefundRequest;
+import me.performancereservation.domain.refund.enums.RefundStatus;
+import me.performancereservation.global.exception.AppException;
+import me.performancereservation.global.exception.ErrorCode;
+import me.performancereservation.global.exception.ErrorType;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/refunds")
+public class RefundController {
+
+    private final RefundService refundService;
+
+    /*--- USER 요청에 대응 ---*/
+    // 본인 id와 일치하는 모든 환불내역 리스트 반환
+    @GetMapping("/user/{userId}")
+    public ResponseEntity<List<RefundDetailResponse>> getAllRefundDetailsWithUserId(@PathVariable Long userId) {
+        return ResponseEntity.status(HttpStatus.OK)
+                .body(refundService.findAllRefundsDetailByUserId(userId));
+    }
+
+    // 특정 예약id 환불요청 -> 환불내역 생성
+    @PostMapping("/user/new-refund")
+    public ResponseEntity<Long> createNewRefund(@RequestBody RefundRequest refundRequest) {
+        return ResponseEntity.status(HttpStatus.CREATED)
+                .body(refundService.save(refundRequest));
+    }
+
+
+    /*--- ADMIN 요청에 대응 ---*/
+    // 모든 환불 리스트 반환
+    @GetMapping("/admin/all-refunds")
+    public ResponseEntity<List<RefundDetailResponse>> getAllRefundDetails() {
+        return ResponseEntity.status(HttpStatus.OK)
+                .body(refundService.findAllRefundsDetail());
+    }
+
+    // 모든 환불 내역 중 특정 status 리스트 반환
+    @GetMapping("/admin/{refundStatus}")
+    public ResponseEntity<List<RefundDetailResponse>> getAllRefundDetailsByRefundStatus(@PathVariable String refundStatus) {
```

Comment:
```suggestion
    @GetMapping("/admin")
    public ResponseEntity<Page<RefundDetailResponse>> getAllRefundDetailsByRefundStatus(
    @RequestParam(required = false) RefundStatus status,
    @PageableDefault(size = 10, sort = "createdAt", direction = Sort.Direction.DESC) Pageable pageable,
    ) {
```

여기도 좋은데 쿼리 파라미터로 상태 필터링을 적용해보시는건 어떤가요?

/refunds?status=CONFIRMED
/refunds?status=PENDING

아니면 소문자로 받았다가 upperCase 적용시켜서 서비스에서 처리해도 될 것 같구요

Reply:
string으로 받았다가 upperCase 적용시키고 Status로 변환하는 것이 좋아보입니다! 
그리고 required = false로 두신 걸 보고 떠올랐는데, 모두 /admin으로 get 요청을 받되
status = null이면 전체 조회, 
status = PENDING혹은 CONFIRM이면 각 상태에 맞는 내역 조회로 합치는 건 어떨까요?
현재는 `getAllRefundDetails`와 `getAllRefundDetailsByRefundStatus`로 분리되어 있습니다. 찾아보니 request param 유무로는 엔드포인트를 구분하지 못해 에러가 발생한다고 한다고 해서요..

Reply:
말씀해주신 것처럼 그게 좋아보여요!